api/users/views.py

```diff
@@ -18,11 +21,16 @@ class UserMixin(object):
 
     def get_user(self, check_permissions=True):
         key = self.kwargs[self.node_lookup_url_kwarg]
+        current_user = self.request.user
 
         if key == 'me':
-            return self.request.user
+            if isinstance(current_user, AnonymousUser):
+                raise NotFound
```

Comment:
We went with a 404 because it's in the path, although… I think it also uses this when it's passed as a parameter as well. Hmm.